tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersIndexKeys.cs

```diff
@@ -0,0 +1,95 @@
+            _ = Builders<Person>.IndexKeys.Descending("Address");
+        }
+
+        [BuildersMQL("{ \"Address\" : -1, \"Name\" : 1, \"LastName\" : -1 }")]
+        [BuildersMQL("{ \"SiblingsCount\" : 1, \"TicksSinceBirth\" : -1, \"Vehicle\" : 1 }")]
+        public void Ascending_descending()
+        {
+            _ = Builders<Person>.IndexKeys.Descending(x => x.Address).Ascending("Name").Descending(x => x.LastName);
+            _ = Builders<Person>.IndexKeys.Ascending(x => x.SiblingsCount).Descending("TicksSinceBirth").Ascending(x => x.Vehicle);
+        }
+
+        [BuildersMQL("{ \"Vehicle\" : \"2d\" }")]
+        [BuildersMQL("{ \"Address\" : \"2d\", \"Vehicle\" : \"2d\" }")]
+        public void Geo2D()
+        {
+            _ = Builders<Person>.IndexKeys.Geo2D(u => u.Vehicle);
+            _ = Builders<Person>.IndexKeys.Geo2D(u => u.Address).Geo2D("Vehicle");
+        }
+
+        [BuildersMQL("{ \"Vehicle\" : \"2dsphere\" }")]
+        [BuildersMQL("{ \"Address\" : \"2dsphere\", \"Vehicle\" : \"2dsphere\" }")]
+        public void Geo2DSphere()
+        {
+            _ = Builders<Person>.IndexKeys.Geo2DSphere(u => u.Vehicle);
+            _ = Builders<Person>.IndexKeys.Geo2DSphere(u => u.Address).Geo2DSphere("Vehicle");
+        }
+
+        [BuildersMQL("{ \"Vehicle\" : \"geoHaystack\" }")]
+        [BuildersMQL("{ \"Address\" : \"geoHaystack\", \"Vehicle\" : \"geoHaystack\" }")]
+        public void GeoHaystack()
+        {
+            _ = Builders<Person>.IndexKeys.GeoHaystack(u => u.Vehicle);
+            _ = Builders<Person>.IndexKeys.GeoHaystack(u => u.Address).GeoHaystack("Vehicle");
+        }
+
+        [BuildersMQL("{ \"Vehicle\" : \"hashed\" }")]
+        [BuildersMQL("{ \"Address\" : \"hashed\", \"Vehicle\" : \"hashed\" }")]
+        public void Hashed()
+        {
+            _ = Builders<Person>.IndexKeys.Hashed(u => u.Vehicle);
+            _ = Builders<Person>.IndexKeys.Hashed(u => u.Address).Hashed("Vehicle");
+        }
+
+        [BuildersMQL("{ \"Vehicle\" : \"text\" }")]
+        [BuildersMQL("{ \"Address\" : \"text\", \"Vehicle\" : \"text\" }")]
+        public void Text()
+        {
+            _ = Builders<Person>.IndexKeys.Text(u => u.Vehicle);
+            _ = Builders<Person>.IndexKeys.Text(u => u.Address).Text("Vehicle");
+        }
+
+        [BuildersMQL("{ \"Name\" : \"text\", \"LastName\" : \"hashed\", \"Vehicle\" : \"2d\", \"Address\" : 1 }")]
+        [BuildersMQL("{ \"Name\" : \"2d\", \"LastName\" : \"2dsphere\", \"Vehicle\" : \"text\", \"Address\" : -1, \"SiblingsCount\" : \"hashed\", \"TicksSinceBirth\" : 1 }")]
+        public void Combined()
+        {
+            _ = Builders<Person>.IndexKeys.Text(u => u.Name).Hashed(u => u.LastName).Geo2D(u => u.Vehicle).Ascending(u => u.Address);
+            _ = Builders<Person>.IndexKeys.Geo2D(u => u.Name).Geo2DSphere(u => u.LastName).Text(u => u.Vehicle).Descending(u => u.Address).Hashed(u => u.SiblingsCount).Ascending(u => u.TicksSinceBirth);
```

Comment:
Add also a test for IndexKeysDefinition<TDocument>.Combine

Reply:
Done